test/goHome.test.ts

```diff
@@ -0,0 +1,79 @@
+import { chromium, Page, Browser, BrowserContext } from "playwright"
+
+describe("login", () => {
+  let browser: Browser
+  let page: Page
+  let context: BrowserContext
+
+  beforeAll(async (done) => {
+    browser = await chromium.launch()
+    // Create a new context with the saved storage state
+    const storageState = JSON.parse(process.env.STORAGE || "")
+    context = await browser.newContext({ storageState, recordVideo: { dir: "./test/videos/" } })
+    done()
+  })
+
+  afterAll(async (done) => {
+    // Remove password from local storage
+    await context.clearCookies()
+
+    await browser.close()
+    await context.close()
+    done()
+  })
+
+  beforeEach(async (done) => {
+    page = await context.newPage()
+    done()
+  })
+
+  it("should see a 'Go Home' button in the Application Menu that goes to /healthz", async (done) => {
+    const GO_HOME_URL = `${process.env.CODE_SERVER_ADDRESS}/healthz`
+    let requestedGoHomeUrl = false
+    page.on("request", (request) => {
+      // This ensures that we did make a request to the GO_HOME_URL
+      // Most reliable way to test button
+      // because we don't care if the request has a response
+      // only that it was made
+      if (request.url() === GO_HOME_URL) {
+        requestedGoHomeUrl = true
+        expect(requestedGoHomeUrl).toBeTruthy()
+
+        // This ensures Jest knows we're done here.
+        done()
+      }
+    })
+    // Sometimes a dialog shows up when you navigate
+    // asking if you're sure you want to leave
+    // so we listen if it comes, we accept it
+    page.on("dialog", (dialog) => dialog.accept())
+
+    // waitUntil: "domcontentloaded"
+    // In case the page takes a long time to load
+    await page.goto(process.env.CODE_SERVER_ADDRESS || "http://localhost:8080", { waitUntil: "domcontentloaded" })
+
+    // For some odd reason, the login method used in globalSetup.ts
+    // I don't know if it's on playwright clearing our cookies by accident
+    // or if it's our cookies disappearing.
```

Comment:
One thing we could try is setting the cookie using the password directly. So something like (untested, the syntax might be a bit off and I'm not sure if we'd need to set some other params like `url`, `domain`, etc):

```
import { hash } from "../src/node/util"

browser.newContext({ storageState: { cookies: [{ name: "key", value: hash(process.env.PASSWORD) }]} })
```

Still, definitely weird it's not working.

Reply:
I'll try this out!

Reply:
I think that worked! I ran the test locally and it passed lol